test/clj/rems/api/test_forms.clj

```diff
@@ -61,32 +61,75 @@
                        :id)]
             (is id)
             (testing "and fetch"
-              (let [form (-> (request :get (str "/api/forms/" id))
-                             (authenticate api-key user-id)
-                             handler
-                             read-ok-body)
-                    form-template (-> (request :get (str "/api/forms/v2/" id))
+              (let [form-template (-> (request :get (str "/api/forms/" id))
                                       (authenticate api-key user-id)
                                       handler
                                       read-ok-body)]
-                (testing "v2 result matches input"
+                (testing "result matches input"
                   (is (= (select-keys command [:title :organization])
                          (select-keys form-template [:title :organization])))
-                  (is (= (:items command)
-                         (:fields form-template))))
-                (testing "v1 result matches input"
-                  (is (= (select-keys command [:title :organization])
-                         (select-keys form [:title :organization])))
-                  (is (= (:items command)
-                         (:fields (form/get-form-template (:id form))))))))))))))
+                  (is (= (:fields command)
+                         (:fields form-template))))))))))))
+
+(deftest forms-api-all-field-types-test
+  (let [api-key "42"
+        user-id "owner"
+        ;;"attachment" "date" "description" "label" "multiselect" "option" "text" "texta"
+        localized {:en "en" :fi "fi"}
+        form-spec {:organization "abc" :title "all field types test"
+                   :fields [{:type "text"
+                             :title localized
+                             :optional false}
+                            {:type "texta"
+                             :title localized
+                             :optional true
+                             :maxlength 300
+                             :input-prompt localized}
+                            {:type "description"
+                             :title localized
+                             :optional false}
+                            {:type "option"
+                             :title localized
+                             :optional true
+                             :options [{:key "true" :label localized}
+                                       {:key "false" :label localized}]}
```

Comment:
Option fields can have more than two options. The difference between option and multiselect is that option lets the user choose only one value, but multiselect lets the user choose many values. To make that explicit, this example could be:
```clj
:options [{:key "a" :label localized}
          {:key "b" :label localized}
          {:key "c" :label localized}]